src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2325,6 +2321,23 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
+            case NI_Sve_Scatter16BitWithByteOffsetsNarrowing:
+            case NI_Sve_Scatter32BitWithByteOffsetsNarrowing:
+            case NI_Sve_Scatter8BitWithByteOffsetsNarrowing:
+            case NI_Sve_ScatterWithByteOffsets:
```

Comment:
Were there any other instructions (such as `gather`) which differed between indices vs offsets like these ones?

I know we've reviewed/approved a few groups, but I don't know which are actually implemented today.

Reply:
I found the following APIs that are incorrectly treating offsets as indices and would need a similar fix.

```
GatherVectorInt16WithByteOffsetsSignExtend (long/ulong combinations)
GatherVectorInt32WithByteOffsetsSignExtend (long/ulong)
GatherVectorInt16WithByteOffsetsSignExtendFirstFaulting (long/ulong)
GatherVectorInt32WithByteOffsetsSignExtendFirstFaulting (long/ulong)
GatherVectorUInt16WithByteOffsetsZeroExtend (long/ulong)
GatherVectorUInt16WithByteOffsetsZeroExtendFirstFaulting (long/ulong)
GatherVectorUInt32WithByteOffsetsZeroExtend(long/ulong)
```

Do you recommend a separate PR for the above or should put in the current one?

Reply:
Since its the same general issue and we're looking at backporting, I'd put them in the same PR

Reply:
We should make sure these ones still get a fix done and also included in the backport